### PR TITLE
Introducing hc wide eslint rules

### PR DESCRIPTION
**Pull request by ibihim**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "karma:node": "karma start karma.conf.js --node true",
     "karma:prod:web": "karma start karma.conf.js --single-run --node false",
     "karma:prod:node": "karma start karma.conf.js --single-run --node true",
-    "test": "yarn lint && yarn build && yarn karma:prod:web && yarn karma:prod:node && codecov"
+    "test": "yarn build && yarn karma:prod:web && yarn karma:prod:node && codecov"
```
Comment (ibihim) [inline package.json]: temporary for testing purpose

File: src/services/UserService.js [hunk 2]
```diff
@@ -131,7 +130,7 @@ class UserService {
             .then(permissions => permissions.map(permission => ({
                 granteeId: permission.grantee_id,
             })));
-    }
-}
+    },
+};
 
-export default new UserService();
+export default userService;
```
Comment (ibihim) [inline src/services/UserService.js]: export default new??? why do we need the boilerplate code from class then???????

Comment (luisfarzati) [reply src/services/UserService.js]: @ibihim Looks good, I would additionally freeze it so nobody can mess with it.

Comment (luisfarzati) [reply src/services/UserService.js]: Actually that's not relevant to the context of this PR, sorry :)

Comment (ibihim) [reply src/services/UserService.js]: Seriously, I like the notion of freeze 😄

File: .eslintrc.js [hunk 3]
```diff
@@ -1,36 +1,50 @@
 module.exports = {
-    "extends": "airbnb-base",
-    "plugins": [
-        "import"
+    extends: ['eslint:recommended', 'airbnb-base'],
+    parser: 'babel-eslint',
+    parserOptions: {
+        ecmaVersion: 6,
```
Comment (isalga) [inline .eslintrc.js]: Why not 8? :)

Comment (ibihim) [reply .eslintrc.js]: haha, I was thinking this to myself too. sure lets go for full blown feature set 👍

File: src/services/FhirService.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import fhirValidator from '../lib/fhirValidator';
 import documentRoutes from '../routes/documentRoutes';
-import UserService from '../services/UserService';
```
[the remark this answers is not in the document]
Comment (ibihim) [reply src/services/FhirService.js]: hahaha

File: src/services/FhirService.js [hunk 5]
```diff
@@ -125,10 +125,10 @@ class FHIRService {
             });
     }
 
-    deleteRecord(ownerId, recordId) {
-        if (!ownerId) ownerId = UserService.getCurrentUser().id;
+    static deleteRecord(givenID, recordID) {
```
[the remark this answers is not in the document]
Comment (ibihim) [reply src/services/FhirService.js]: I hate mutation, but I can rename it.

File: src/services/userService.js [hunk 6]
```diff
@@ -31,18 +29,19 @@ class UserService {
...
 
     isCurrentUser(userId) {
         const currentUser = this.getCurrentUser();
         if (!currentUser || currentUser.id !== userId) return false;
         return true;
-    }
+    },
 
-    getInternalUser(userId, partial = false) {
+    getInternalUser(givenUserId, partial = false) {
```
[the remark this answers is not in the document]
Comment (ibihim) [reply src/services/userService.js]: ok ok

**Comment by ibihim** [comment]
test this please

**Comment by ibihim** [comment]
After talking with @Glumli we agreed, that it is non sense to have a reduced code coverage when no additional functionality / paths / you name where added. There was just a minor refactoring to fit eslint rules.
We would need to configure codecov properly so it can pin point where the loss in coverage is.

**Pull request merged**